core/store/models/log_events.go

```diff
@@ -85,20 +85,25 @@ func TopicFiltersForRunLog(logTopics []common.Hash, jobID *ID) [][]common.Hash {
 }
 
 // FilterQueryFactory returns the ethereum FilterQuery for this initiator.
-func FilterQueryFactory(i Initiator, from *big.Int) (ethereum.FilterQuery, error) {
+func FilterQueryFactory(i Initiator, nextHead *big.Int) (ethereum.FilterQuery, error) {
 	q := ethereum.FilterQuery{
-		FromBlock: from,
+		FromBlock: nextHead,
 		Addresses: utils.WithoutZeroAddresses([]common.Address{i.Address}),
 	}
 
 	switch i.Type {
 	case InitiatorEthLog:
-		if from == nil {
+		if nextHead == nil {
 			q.FromBlock = i.InitiatorParams.FromBlock.ToInt()
-		} else if from != nil && i.InitiatorParams.FromBlock != nil {
-			q.FromBlock = utils.MaxBigs(from, i.InitiatorParams.FromBlock.ToInt())
+		} else if nextHead != nil && i.InitiatorParams.FromBlock != nil {
+			q.FromBlock = utils.MaxBigs(nextHead, i.InitiatorParams.FromBlock.ToInt())
 		}
 		q.ToBlock = i.InitiatorParams.ToBlock.ToInt()
+
+		if q.FromBlock != nil && q.ToBlock != nil && q.FromBlock.Cmp(q.ToBlock) >= 0 {
+			return q, fmt.Errorf("cannot generate a FilterQuery with fromBlock >= toBlock")
```

Comment:
Have you seen if geth/parity blow up in this case? It would be nice to punt and leave it up to them. I think you could get in a situation where you want to jump ahead of some height and throwing an error would prevent us from doing that. It looks like the node will still start, and the error will prevent this job from running, which is essentially the behavior we want... right?

Reply:
Going to assume erroring is ok for now.